src/main/java/com/example/bookstore/controller/CategoryController.java

```diff
@@ -0,0 +1,78 @@
+package com.example.bookstore.controller;
+
+import com.example.bookstore.dto.book.BookDtoWithoutCategoryIds;
+import com.example.bookstore.dto.category.CategoryDto;
+import com.example.bookstore.dto.category.CreateCategoryRequestDto;
+import com.example.bookstore.service.BookService;
+import com.example.bookstore.service.CategoryService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Category management", description = "Endpoints for managing categories")
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/category")
```

Comment:
According to task description and REST naming convention
```suggestion
@RequestMapping("/api/categories")
```

Reply:
Oops :)